coap-gateway/service/session.go

```diff
@@ -93,21 +95,18 @@ func (a *authorizationContext) ToContext(ctx context.Context) context.Context {
 
 // session a setup of connection
 type session struct {
-	server        *Service
-	coapConn      mux.Conn
-	tlsDeviceID   string
-	tlsValidUntil time.Time
-
-	resourceSubscriptions *kitSync.Map // [token]
-
-	exchangeCache *ExchangeCache
-	refreshCache  *RefreshCache
-
-	mutex                   sync.Mutex
+	tlsValidUntil           time.Time
+	coapConn                mux.Conn
+	server                  *Service
+	resourceSubscriptions   *kitSync.Map
+	exchangeCache           *ExchangeCache
+	refreshCache            *RefreshCache
 	authCtx                 *authorizationContext
 	deviceSubscriber        *grpcClient.DeviceSubscriber
 	deviceObserver          *future.Future
 	closeEventSubscriptions func()
+	tlsDeviceID             string
+	mutex                   sync.Mutex
```

Comment:
mutex guards `authCtx`, `deviceSubscriber`, `deviceObserver` and `closeEventSubscriptions`

Reply:
moved to private